### DISCUSSION TITLE
Jest integration nested values

### DISCUSSION THREAD

**Discussion by alexnsorensen** [prisma/prisma#24102] [Q&A] [answered]
### Question

I'm using Jest and jest-mock-extended to do unit testing on some more complex functions in my code. However I ran into an issue where I need to get nested data or something from prisma. and I can't see anything on [Prisma's official website](https://www.prisma.io/docs/orm/prisma-client/testing/unit-testing) about this and couldn't see anything, I also had a read through [this article](https://dev.to/eddeee888/how-to-write-tests-for-prisma-with-docker-and-jest-593i)

For example:
```js
primsa.person.findFirstOrThrow({where: {id: 1}}).posts()`
```

When I just run this code: 
```js
primsa.person.findFirstOrThrow({where: {id: 1}})
```
Jest works great and gets the mocked data, but when I add the .posts() I get an error saying **TypeError: person.posts is not a function**.

I do my mocked person data like so:

```js
mockPrisma.person.findUniqueOrThrow.mockResolvedValue({
    id: 1
    name: 'John Smith'
});
```
I understand why this error occurs, but I can't work out the syntax to get the mocked posts in there as everything I try throws a TS error.

Things I tried (assume data between the []):

```js
mockPrisma.person.findUniqueOrThrow.mockResolvedValue({
    id: 1
    name: 'John Smith',
    posts: mockPrisma.post.findMany.mockResolvedValue([]);
});

////////////////////////////////////////////////////

mockPrisma.post.findMany.mockResolvedValue([])

mockPrisma.person.findUniqueOrThrow.mockResolvedValue({
    id: 1
    name: 'John Smith',
    posts: mockPrisma.post.findMany
});
```

**Comment by ludralph** [comment]
Hi @alexnsorensen 👋 

Thank you for raising this question.

Can you use `mockReturnValue` for the first call instead of `mockResolvedValue`?

```js
prismaMock.person.findUniqueOrThrow.mockReturnValue({
    posts: jest.fn().mockResolvedValue(posts),
});
```

**Reply by alexnsorensen** [reply, marked as answer]
Hi, what I ended up having to do was this:

```
prismaMock.person.findUniqueOrThrow.mockReturnValue({
    posts: jest.fn().mockResolvedValue(posts),
    then: jest.fn().mockReturnValue(Promise.resolve(mockedPerson))(),
    catch: jest.fn().mockResolvedValue(null),
    finally: jest.fn().mockResolvedValue(mockedPerson)(),
    [Symbol.toStringTag]: 'PrismaPromise',
});
```

If I didn't add these then I would get a tonne of TS errors, so I Imagine it's to do with my TS settings probably being rather strict...

**Reply by raphaelumoh** [reply]
Thank you for sharing the final outcome. It's really helpful.
